apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs

```diff
@@ -413,26 +418,81 @@ private LogEntry BuildLogEntry(LogEventInfo loggingEvent)
+                jsonStruct.Fields.Add("properties", Value.ForStruct(propertiesStruct));
+
+                foreach (var combinedProperty in GetAllProperties(loggingEvent))
                 {
-                    logEntry.Labels[combinedProperty.Key] = combinedProperty.Value?.ToString() ?? "null";
+                    if (string.IsNullOrEmpty(combinedProperty.Key))
+                        continue;
+
+                    try
+                    {
+                        switch (Convert.GetTypeCode(combinedProperty.Value))
+                        {
+                            case TypeCode.Empty:
+                                propertiesStruct.Fields.Add(combinedProperty.Key, Value.ForNull());
+                                break;
+                            case TypeCode.Boolean:
+                                propertiesStruct.Fields.Add(combinedProperty.Key, Value.ForBool((bool)combinedProperty.Value));
+                                break;
+                            case TypeCode.Decimal:
+                            case TypeCode.Double:
+                            case TypeCode.Single:
+                            case TypeCode.Byte:
+                            case TypeCode.SByte:
+                            case TypeCode.Int16:
+                            case TypeCode.UInt16:
+                            case TypeCode.Int32:
+                            case TypeCode.UInt32:
+                            case TypeCode.Int64:
+                            case TypeCode.UInt64:
+                                propertiesStruct.Fields.Add(combinedProperty.Key, Value.ForNumber(Convert.ToDouble(combinedProperty.Value)));
+                                break;
+                            default:
+                                propertiesStruct.Fields.Add(combinedProperty.Key, Value.ForString(combinedProperty.Value?.ToString() ?? "null"));
+                                break;
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        InternalLogger.Warn(ex, "GoogleStackdriver(Name={0}): Exception at BuildLogEntry with Key={1}", Name, combinedProperty.Key);
+                        propertiesStruct.Fields.Add(combinedProperty.Key, Value.ForNull());
+                    }
                 }
-                catch (Exception ex)
+
+                logEntry.JsonPayload = jsonStruct;
+            }
+            else
+            {
+                logEntry.TextPayload = RenderLogEvent(Layout, loggingEvent);
+
+                foreach (var combinedProperty in GetAllProperties(loggingEvent))
                 {
-                    InternalLogger.Warn(ex, "GoogleStackdriver(Name={0}): Exception at BuildLogEntry with Key={1}", Name, combinedProperty.Key);
-                    logEntry.Labels[combinedProperty.Key] = "null";
+                    if (string.IsNullOrEmpty(combinedProperty.Key))
+                        continue;

```

Comment:
Please use braces around `continue`:
```C#
if (string.IsNullOrEmpty(combinedProperty.Key))
{
    continue;
}
```

I know this makes this `if` statement more verbose, but our coding standard is to always use braces.

Reply:
Fixed